atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -133,14 +134,13 @@ def is_image_build_type(self, base_image):
         return base_image.strip().lower() == 'koji/image-build'
 
     def extract_base_url(self, repo_url):
-        session = get_retrying_requests_session()
-        response = session.get(repo_url)
-        response.raise_for_status()
-        repo = ConfigParser()
-        repo.readfp(StringIO(response.text))
-
-        return [repo.get(section, 'baseurl') for section in repo.sections()
-                if repo.has_option(section, 'baseurl')]
+        yum_repo = YumRepo(repo_url)
+        yum_repo.fetch()
+        if yum_repo.is_valid():
+            repo = yum_repo.config
+            return [repo.get(section, 'baseurl') for section in repo.sections()
+                    if repo.has_option(section, 'baseurl')]
+        return []
```

Comment:
Minor: I'd slightly prefer to have the "happy path" at the top level, e.g.:
```
if not yum_repo.is_valid():
    return []

repo = ...
return [...]
```

Reply:
Fixed.